malaria-ios/ViewControllers/UserProfileViewController.swift

```diff
@@ -0,0 +1,199 @@
+    tableViewHeightConstraint.constant = CGFloat(medicine.count) * Constants.UserProfile.CellHeightAndOffset
+  }
+}
+
+// MARK: IBActions and helpers
+
+extension UserProfileViewController {
+  
+  @IBAction func settingsBtnHandler(sender: AnyObject) {
+    dispatch_async(dispatch_get_main_queue()) {
+      let view = UIStoryboard.instantiate(viewControllerClass: SetupScreenViewController.self)
+      view.delegate = self
+      self.presentViewController(view, animated: true, completion: nil)
+    }
+  }
+  
+  @IBAction func remindMeWeeksPressed(sender: UIButton) {
+    let alertController = UIAlertController(title: "Set reminder for:", message: nil, preferredStyle: UIAlertControllerStyle.Alert)
+    
+    for intervalValue in PillStatusNotificationsManager.ReminderInterval.allValues {
+      let action = UIAlertAction(title: intervalValue.toString(),
+                                 style: UIAlertActionStyle.Default,
+                                 handler: {
+                                  (alert: UIAlertAction!) in
+                                  
+                                  self.reminderValue = intervalValue
+                                  
+                                  // Save the reminder value internally
+                                  UserSettingsManager.UserSetting.PillReminderValue.setString(self.reminderValue.toString())
+                                  
+                                  // Recalculate medicine left
+                                  self.calculateMedicineLeft()
+      })
+      
+      alertController.addAction(action)
+    }
+    
+    presentViewController(alertController, animated: true, completion: nil)
+  }
+}
+
+// MARK: Table View Data Source
+extension UserProfileViewController: UITableViewDataSource {
+  
+  func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    return medicine.count
+  }
+  
+  func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+    let cell = tableView.dequeueReusableCellWithIdentifier(Constants.UserProfile.CellReuseIdentifier, forIndexPath: indexPath) as! UserProfilePillTableViewCell
+    
+    let name = medicine[indexPath.row].name
+    let remainingMedicine = medicine[indexPath.row].remainingMedicine
+    
+    cell.updateCellWithParameters(self,
+                                  name: name,
+                                  remainingMedicine: String(remainingMedicine),
+                                  indexPath: indexPath)
+    cell.delegate = self
+    
+    return cell
+  }
+}
+
+// MARK: Table View Delegate
+
+extension UserProfileViewController: UITableViewDelegate {
+  
+  // Solves iPad white cells bug:
+  // http://stackoverflow.com/questions/24977047/tableview-cell-on-ipad-refusing-to-accept-clear-color
+  
+  func tableView(tableView: UITableView, willDisplayCell cell: UITableViewCell, forRowAtIndexPath indexPath: NSIndexPath) {
+    
```

Comment:
Which line? Do I not know how to read comments? :))

I see multiple lines marked. Probably the link.